plugin/helpers.go

```diff
@@ -6,25 +6,26 @@ package plugin
 import "github.com/mattermost/mattermost-server/model"
 
 type Helpers interface {
-	// EnsureBot either returns an existing bot user or creates a bot user with
-	// the specifications of the passed bot.
-	// Returns the id of the bot created or existing.
+	// EnsureBot either returns an existing bot user matching the given bot, or creates a bot user from the given bot.
+	// Returns the id of the resulting bot.
 	EnsureBot(bot *model.Bot) (string, error)
 
-	// KVGetJSON retrievs a value based on the key.
-	KVGetJSON(key string, value interface{}) error
-
-	// KVSetJSON stores a key-value pair.
+	// KVSetJSON stores a key-value pair, unique per plugin, marshalling the given value as a JSON string.
 	KVSetJSON(key string, value interface{}) error
 
-	// KVCompareAndSetJSON updates a key-value pair if the current
-	// value is equal to oldValue.
-	// Returns (false, err) if DB/marshal/unmarshal error occurred
-	// Returns (false, nil) if current value != old value
-	// Returns (true, nil) if current value == old value or new key is inserted
+	// KVCompareAndSetJSON updates a key-value pair, unique per plugin, but only if the current value matches the given oldValue after marshalling as a JSON string.
+	// Inserts a new key if oldValue == nil.
+	// Returns (false, err) if DB error occurred
+	// Returns (false, nil) if current value != oldValue or key already exists when inserting
+	// Returns (true, nil) if current value == oldValue or new key is inserted
+	//
+	// Minimum server version: 5.12
```

Comment:
Why do you want to document this? For plugin using the method it's pretty clear, if they are present.

Reply:
`KVCompareAndSetJSON` only works with a server that supports `KVCompareAndSet`, i.e. 5.12. It's entirely possible for the plugin to have been compiled against a newer version of the package, while still pointing at an older server.

Looks like I forgot to be explicit about `KVSetWithExpiryJSON`, but the others follow the pattern of not being explicit since that functionality has been present since the beginning.

Reply:
Makes sense. Thanks for the clarification.